Cancel tts on feature to other pages and added check for empty steps/ingredients

tnl001 opened the pull request:
Resolves #141 , Resolves #129

Changelog (11/30/2021):

- Trigger TTS only for the correct children (view-my-recipe or view-fea-recipe) of view-recipe-page
- Added a check for empty instruction steps
- Added a check for empty ingredient

danielpak1 reviewed (changes requested):
tts cancel only works when you click "back" or "nutrition" 

You need to somehow add speech synthesis canceling when navigating back to the landing page via escape or clicking the logo

tnl001 commented:
Changelog:
- Trigger TTS only for the correct children (view-my-recipe or view-fea-recipe) of view-recipe-page
- Added a check for empty instruction steps
- Added a check for empty ingredient
